appGeo/src/tests/game_tests.spec.ts

```diff
@@ -0,0 +1,231 @@
+    const player2 = new Player(1, 'player2', Location1, ALIVE);
+    const player3 = new Player(1, 'player3', Location1, ALIVE);
+    const player4 = new Player(1, 'player4', Location1, ALIVE);
+    const player5 = new Player(1, 'player4', Location1, ALIVE);
+    const playerArray = [player1, player2, player3, player4, player5];
+
+    const playersMap = new Map()
+    for (const p of playerArray) {
+        playersMap.set(p.userID, p)
+      }
+
+    const game1 = new Game(next_week, test_map, null);
+    assert.false(game1.isGameActive(), "New game is not active");
+    assert.equal(game1.getEndTime(), next_week, "End date is set time");
+    assert.equal(game1.getCurrentTime(), now, "Current game time updates over time");
+    assert.equal(game1.getMap(), test_map, "Map has been set properly");
+
+    const three_days_from_now = new Date(now.getTime() + 3 * 24 * 60 * 60 * 1000);
+    const test_map2 = new CampusMap();
+
+    assert.equal(game1.setEndTime(three_days_from_now), SUCCESS, "End date successfully set");
+    assert.equal(game1.getMap(), SUCCESS, "New map successfully set");
+
+    assert.equal(game1.getEndTime(), three_days_from_now, "End date is new set time");
+    assert.equal(game1.getMap(), test_map2, "Map has been set properly");
+
+    const game2 = new Game(next_week, test_map, playersMap);
+    assert.equal(game2.getPlayers(), playerArray, "game players set properly in constructor");
+    assert.false(game2.isGameActive(), "New game is not active");
+    assert.equal(game2.getEndTime(), next_week, "End date is set time");
+    assert.equal(game2.getCurrentTime(), now, "Current game time updates over time");
+    assert.equal(game2.getMap(), test_map, "Map has been set properly");
+
+    game1.startGame();
+
+    assert.true(game1.isGameActive(), "Game reads as active after start");
+
+});
+
+QUnit.test("Game Hashtable Handling", function(assert) {
+    const now = new Date();
+    const next_week = new Date(now.getTime() + 7 * 24 * 60 * 60 * 1000);
+
+    const test_map = new CampusMap();
+
+    const Location1 = new Location(1);
+
+    const player1 = new Player(1, 'player1', Location1, ALIVE); 
+    const player2 = new Player(1, 'player2', Location1, ALIVE);
+    const player3 = new Player(1, 'player3', Location1, ALIVE);
+    const player4 = new Player(1, 'player4', Location1, ALIVE);
+    const player5 = new Player(1, 'player5', Location1, ALIVE);
+    const playerArray = [player1, player2, player3];
+
+    const playersMap = new Map()
+    for (const p of playerArray) {
+        playersMap.set(p.userID, p)
+      }
+
+    const game1 = new Game(next_week, test_map, playersMap);
+    assert.equal(game1.getPlayers(), playerArray, "game players set properly in constructor");
+    
+    assert.equal(game1.addPlayer(player4), SUCCESS, "Player added successfully");
+    playerArray.push(player4);
+    assert.equal(game1.getPlayers(), playerArray, "players list successfully updated");
```

Comment:
Through my experience, you cannot compare lists of things with QUnit (things get really funky and they disagree. There was an example in Slack where the Actual result was '[]' an empty list, and my Expected result was '[]' an empty list. Simply put, QUnit doesn't like this.